### PR TITLE
feat(modal): add title slot and close slot

### PR DESCRIPTION
**Pull request by wzh425**

### CODE REVIEW

File: src/Masa.Blazor/Presets/Modal/ModalBase.razor [hunk 1]
```diff
@@ -62,7 +76,7 @@
                         else
                         {
                             <ModalButton Props="ComputedDeleteButtonProps"
-                                         OnClick="HandleOnDelete">
+                             OnClick="HandleOnDelete">
```
Comment (capdiem) [inline src/Masa.Blazor/Presets/Modal/ModalBase.razor:79]: rollback this line.

File: src/Masa.Blazor/Presets/Modal/ModalBase.razor [hunk 2]
```diff
@@ -79,8 +93,8 @@
                         else
                         {
                             <ModalButton Props="ComputedCancelButtonProps"
-                                         Disabled="_saveLoading"
-                                         OnClick="HandleOnCancel">
+                             Disabled="_saveLoading"
+                             OnClick="HandleOnCancel">
```
Comment (capdiem) [inline src/Masa.Blazor/Presets/Modal/ModalBase.razor:97]: rollback these lines.

File: src/Masa.Blazor/Presets/Modal/ModalBase.razor [hunk 3]
```diff
@@ -95,9 +109,9 @@
                         else
                         {
                             <ModalButton Props="ComputedSaveButtonProps"
-                                         Disabled="_saveLoading"
-                                         Loading="_saveLoading"
-                                         OnClick="HandleOnSave">
+                             Disabled="_saveLoading"
+                             Loading="_saveLoading"
+                             OnClick="HandleOnSave">
```
Comment (capdiem) [inline src/Masa.Blazor/Presets/Modal/ModalBase.razor:114]: rollback these lines.

File: src/Masa.Blazor/Presets/Modal/ModalBase.razor.cs [hunk 4]
```diff
@@ -126,6 +126,12 @@ public partial class ModalBase
         [Parameter]
         public RenderFragment<(Func<MouseEventArgs, Task> Click, bool Loading)> CancelContent { get; set; }
 
+        [Parameter]
+        public RenderFragment TitleContent { get; set; }
+
+        [Parameter]
+        public RenderFragment<(Func<MouseEventArgs, Task> Click, bool Loading)> CloseContent { get; set; }
```
Comment (capdiem) [inline src/Masa.Blazor/Presets/Modal/ModalBase.razor.cs:133]: Loading is not needed.

**Review by capdiem — CHANGES_REQUESTED** [review]

**Review by capdiem — APPROVED** [review]
LGTM. @doddgu

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**